src/app/components/take-poll/take-poll.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { TakePollService } from './take-poll.service';
+
+@Component({
+  selector: 'app-take-poll',
+  templateUrl: './take-poll.component.html',
+  styleUrls: ['./take-poll.component.css']
+})
+export class TakePollComponent implements OnInit {
+  loader: boolean;
+  pollList: any;
+  takePollForm: FormGroup;
+  voted: Array<string> = [];
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private takePollService: TakePollService) { }
+
+  ngOnInit() {
+    this.voted = JSON.parse(localStorage.getItem("voted")) || [];
+    this.viewPoll();
```

Comment:
we can chain this function as i can see here.
we don't really need a form unless we fetch the data. so we do it like 

viewPoll() {
...
createTakePollForm()
}